### ISSUE TITLE
JVM crashes with malloc: *** error with not default video codecs

### ISSUE THREAD

**Issue by gkozyryatskyy** [bytedeco/javacv#830]
Hello guys.

I have a "bad encoded" video file https://www.dropbox.com/s/3t3dmzpzsvc8aea/Do_not_use_this_movie.mp4?dl=1

When Im executing this code
```
public static void memErrorTest1() throws Exception {
        try (FFmpegFrameGrabber video = new FFmpegFrameGrabber(
                "https://www.dropbox.com/s/3t3dmzpzsvc8aea/Do_not_use_this_movie.mp4?dl=1")) {
            video.start();
            Frame frame;
            while (video.getTimestamp() <= video.getLengthInTime() && (frame = video.grab()) != null) {
                // just iterate
            }
        }
    }
```

Im getting 
```
Exception in thread "main" java.lang.IllegalArgumentException
	at java.nio.Buffer.limit(Buffer.java:275)
	at org.bytedeco.javacv.FFmpegFrameGrabber.processImage(FFmpegFrameGrabber.java:777)
	at org.bytedeco.javacv.FFmpegFrameGrabber.grabFrame(FFmpegFrameGrabber.java:946)
	at org.bytedeco.javacv.FFmpegFrameGrabber.grab(FFmpegFrameGrabber.java:877)
	at com.weavo._test.gk.MainTest.memErrorTest1(MainTest.java:26)
	at com.weavo._test.gk.MainTest.main(MainTest.java:18)
```

But if i adding a FFmpegFrameRecorder there with some, not default, video codec
```
public static void memErrorTest2() throws Exception {
        File out = new File("out.mp4");
        try (FFmpegFrameGrabber video = new FFmpegFrameGrabber(
                "https://www.dropbox.com/s/3t3dmzpzsvc8aea/Do_not_use_this_movie.mp4?dl=1")) {
            video.start();
            try (FFmpegFrameRecorder rec = new FFmpegFrameRecorder(out, video.getImageWidth(), video.getImageHeight(),
                    video.getAudioChannels())) {
                rec.setVideoCodecName("libx264");
                // rec.setVideoCodecName("libopenh264");
                // rec.setVideoCodecName("libx265");
                // rec.setVideoCodecName("mpeg4"); // default
                rec.start();
                Frame frame;
                while (video.getTimestamp() <= video.getLengthInTime() && (frame = video.grab()) != null) {
                    rec.record(frame);
                }
            }
        }
    }
```
Im geting 
```
java(14157,0x70000534c000) malloc: *** error for object 0x1413131313131013: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
```
And my **jvm crashes** with this error. Im getting it with libx264 and libopenh264 codecs.
With libx265 im getting another one 
```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000012ce469e1, pid=14170, tid=0x0000000000005c03
#
# JRE version: Java(TM) SE Runtime Environment (8.0_121-b13) (build 1.8.0_121-b13)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libavcodec.57.dylib+0xc0c9e1]  avpriv_split_xiph_headers+0x330191
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/gleb.kozyryatskyy/work/git/weavo/hs_err_pid14170.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
```
Also with command line ffmpeg I can reencode this video, for example like this 
```
ffmpeg -i Do_not_use_this_movie.mp4 -c:v libx264 -threads:v 4 -profile:v main -level:v 4.0 -c:a copy output.mp4
```
with warnings
```
...
Past duration 0.647987 too large
Past duration 0.647194 too large
Past duration 0.646126 too large
Past duration 0.645058 too large
Past duration 0.643990 too large
Past duration 0.643196 too large
Past duration 0.642128 too large
Past duration 0.641060 too large
Past duration 0.639992 too large
Past duration 0.639198 too large
Past duration 0.638130 too large
Past duration 0.637062 too large
Past duration 0.635994 too large
Past duration 0.635185 too large
Past duration 0.634132 too large
...
```
but still reencoded.

**So the question is:**
 - How can I get rid of this error and reencode the video with libx264 or libopenh264 codec?
 - If i cant get rid of this error how can I, at least, make it catchable  (like with default codec) for not cause jvm (and all the application) crashes?

javacv version
```
        <dependency>
            <groupId>org.bytedeco</groupId>
            <artifactId>javacv-platform</artifactId>
            <version>1.3.3</version>
        </dependency>
```

Thanks a lot!

**Comment by saudet** [comment]
Please try with FFmpeg 3.4.

**Comment by gkozyryatskyy** [comment]
@saudet 
Thank you for the reply!

How can I do this? I mean, how can I change FFmpeg version in javacv-platform build?

**Comment by saudet** [comment]
Add the corresponding "ffmpeg-platform" dependency:
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp-presets/ffmpeg-platform/3.4-1.3.4-SNAPSHOT/
https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/javacpp-presets/ffmpeg/3.4-1.3.4-SNAPSHOT/

**Comment by gkozyryatskyy** [comment]
@saudet Thank you for the reply!
Im add to my pom
```
        <dependency>
            <groupId>org.bytedeco</groupId>
            <artifactId>javacv-platform</artifactId>
            <version>${javacv.version}</version>
            <exclusions>
        		<exclusion>
          			<groupId>org.bytedeco.javacpp-presets</groupId>
          			<artifactId>ffmpeg</artifactId>
        		</exclusion>
        		<exclusion>
          			<groupId>org.bytedeco.javacpp-presets</groupId>
          			<artifactId>ffmpeg-platform</artifactId>
        		</exclusion>
      		</exclusions>
        </dependency>
```
and add 
```
ffmpeg-platform-3.4-1.3.4-20171115.115555-203.jar
ffmpeg-3.4-1.3.4-20171115.115539-203.jar
ffmpeg-3.4-1.3.4-20171115.090437-200-macosx-x86_64.jar
```
to my libraries. 
Have the same errors for all the codecs =(

**Comment by saudet** [comment]
BTW, the video file you provided (https://www.dropbox.com/s/3t3dmzpzsvc8aea/Do_not_use_this_movie.mp4?dl=1) works just fine here with FFmpeg 3.4 and JavaCV 1.3.4-SNAPSHOT. I'm not getting any warnings from FFmpeg and certainly no crash. You will need to provide data to reproduce this issue if we want to have any chance of getting this fixed.

**Comment by gkozyryatskyy** [comment]
@saudet 
try to use 1.3.3 version of JavaCV

**Comment by saudet** [comment]
Why not try it yourself with the SNAPSHOT artifacts?

**Comment by gkozyryatskyy** [comment]
@saudet 
This is exactly what I want to test in the near free time. Thank you for fast reply! Will update on any results.

**Comment by gkozyryatskyy** [comment]
@saudet 
Here is what I did
 - Add to my pom
```
<dependency>
            <groupId>org.bytedeco</groupId>
            <artifactId>javacv-platform</artifactId>
            <version>1.3.3</version>
            <exclusions>
            	<exclusion>
          			<groupId>org.bytedeco</groupId>
          			<artifactId>javacv</artifactId>
        		</exclusion>
        		<exclusion>
          			<groupId>org.bytedeco.javacpp-presets</groupId>
          			<artifactId>ffmpeg</artifactId>
        		</exclusion>
        		<exclusion>
          			<groupId>org.bytedeco.javacpp-presets</groupId>
          			<artifactId>ffmpeg-platform</artifactId>
        		</exclusion>
      		</exclusions>
        </dependency>
```
 - build `javacv-1.3.4.jar` from master
 - add to my libraries
```
javacv-1.3.4.jar
ffmpeg-platform-3.4-1.3.4-20171127.131332-293.jar
ffmpeg-3.4-1.3.4-20171127.131316-293.jar
ffmpeg-3.4-1.3.4-20171127.130859-292-macosx-x86_64.jar
```
And **everything works fine**! (Seems like take more time then alway, but works anyways)

So seems like everything is fixes in 1.3.4.. Waiting for release!

Thank you for your help!
